### ISSUE TITLE
[Question] Help with Network Status Sandbox Permissions

### ISSUE THREAD

**Issue by velten-dev** [flatpak/flatpak#5676]
I have scoured help documentation in Flathub and Flatpak and have not found anything to help resolve an issue I am having. For reference, this application is in Flathub under https://github.com/flathub/finance.reckoner.Reckoner. 

The Reckoner flatpak throws an exception when trying to check the network status. I have been able to package this application as a [Snap](https://codeberg.org/Reckoner/app/src/branch/main/snapcraft.yaml) by adding the plug [network-manager-observe](https://ubuntu.com/core/docs/networkmanager). I have tried following the [sandbox permissions](https://docs.flatpak.org/en/latest/sandbox-permissions.html) documentation and tried adding the [gvfs access](https://docs.flatpak.org/en/latest/sandbox-permissions.html#gvfs-access) to no avail. 

When I try to enable app synchronization, the following exception is thrown. For reference, I am using the [connectivity_plus](https://pub.dev/packages/connectivity_plus) flutter package to check the network connection status. 

```
PLATFORM EXCEPTION: 2024-01-31 09:52:34.502173

SocketException: Connection failed (OS Error: No such file or directory, errno = 2), address = /var/run/dbus/system_bus_socket, port = 0

STACKTRACE:

#0      _NativeSocket.startConnect (dart:io-patch/socket_patch.dart:721)
#1      _NativeSocket.connect (dart:io-patch/socket_patch.dart:984)
#2      _RawSocket.connect (dart:io-patch/socket_patch.dart:1902)
#3      RawSocket.connect (dart:io-patch/socket_patch.dart:21)
#4      DBusClient._openSocket (package:dbus/src/dbus_client.dart:780)
#5      DBusClient._connect (package:dbus/src/dbus_client.dart:800)
#6      DBusClient.callMethod (package:dbus/src/dbus_client.dart:620)
#7      DBusClient.getNameOwner (package:dbus/src/dbus_client.dart:461)
#8      DBusClient._findUniqueName (package:dbus/src/dbus_client.dart:637)
#9      DBusSignalStream._onListen (package:dbus/src/dbus_client.dart:132)
#10     _runGuarded (dart:async/stream_controller.dart:823)
#11     _BroadcastStreamController._subscribe (dart:async/broadcast_stream_controller.dart:207)
#12     _ControllerStream._createSubscription (dart:async/stream_controller.dart:836)
#13     _StreamImpl.listen (dart:async/stream_impl.dart:471)
#14     DBusSignalStream.listen (package:dbus/src/dbus_client.dart:124)
#15     new _ForwardingStreamSubscription (dart:async/stream_pipe.dart:114)
#16     _ForwardingStream._createSubscription (dart:async/stream_pipe.dart:86)
#17     _ForwardingStream.listen (dart:async/stream_pipe.dart:81)
#18     NetworkManagerClient.connect (package:nm/src/network_manager_client.dart:2303)
#19     ConnectivityPlusLinuxPlugin.checkConnectivity (package:connectivity_plus/src/connectivity_plus_linux.dart:24)
#20     Connectivity.checkConnectivity (package:connectivity_plus/connectivity_plus.dart:53)
#21     SyncServiceManager.restart (package:reckoner/controller/sync.dart:163)
<asynchronous suspension>
#22     SyncServiceManager.setSyncData (package:reckoner/controller/sync.dart:108)
<asynchronous suspension>


END EXCEPTION
```

**Comment by smcv** [comment]
> https://github.com/flathub/com.skype.Client/blob/3ed439015f1039088941c50236ec7973248cc4d6/com.skype.Client.json#L16

This is probably a direct equivalent of Snap's `network-manager-observe`. I don't know exactly what `network-manager-observe` does, but from the name, it sounds as though it ought to mean something very similar to "may talk to NetworkManager".

A better long-term solution would be to (ask the app's maintainers to) modify the app to use GLib's [GNetworkMonitor](https://docs.gtk.org/gio/iface.NetworkMonitor.html), or the equivalent in some other runtime library framework like Qt. This would make it more container-friendly, and would also give it better support for systems where NetworkManager is not used. Apps should not normally need to talk directly to NetworkManager unless they are doing network administration.

GNetworkMonitor auto-detects whether it's running inside a sandbox/container or not. If yes (Flatpak, Snap, etc.), it uses https://docs.flatpak.org/en/latest/portal-api-reference.html#gdbus-org.freedesktop.portal.NetworkMonitor. If no, it uses the Linux kernel's netlink interface.

Flatpak seems to be working as intended here, so I'm going to close this as not an actionable issue.

**Comment by smcv** [comment]
Looks like https://github.com/fluttercommunity/plus_plugins/issues/1241 is relevant.

Catching exceptions from checkConnectivity and assuming that the answer is probably "we have connectivity of an unknown type" would probably be a good idea. The best way to detect whether you can reach the Internet is to try to reach the Internet and see what happens!

**Comment by velten-dev** [comment]
Thank you for your help! I will look into these resources.

As a follow-up question, I posted an issue here since I couldn't find a developer help/questions forum or something similar for flatpak/flathub. Was this the correct avenue to ask this question?

I knew this wasn't a platform deficiency and I just needed help on figuring out the confinement issue. I suppose the only thing I missed was searching the plus_plugins repo for a similar issue, but I found the [snap forum for Music Pod](https://forum.snapcraft.io/t/auto-connect-request-network-manager-observe/34461) fix for this and just assumed something similar for flatpak.
